reV/supply_curve/sc_aggregation.py

```diff
@@ -603,60 +616,68 @@ def _get_input_data(gen, gen_fpath, econ_fpath, res_class_dset,
             the extracted arrays from the h5 files.
         """
 
-        if res_class_dset is None:
-            res_data = None
-            res_class_bins = [None]
-        else:
-            res_data = gen[res_class_dset]
+        dset_list = (res_class_dset, cf_dset, lcoe_dset)
+        labels = ('res_class_dset', 'cf_dset', 'lcoe_dset')
+        temp = [None, None, None]
+        for i, dset in enumerate(dset_list):
+            if dset in gen.datasets:
+                temp[i] = gen[dset]
+            else:
+                w = ('Could not find "{}" input as "{}" in '
+                     'generation file: {}. Available datasets: {}'
+                     .format(labels[i], dset, gen_fpath, gen.datasets))
+                logger.warning(w)
+                warn(w, OutputWarning)
 
-        if res_class_bins is None:
+        res_data, cf_data, lcoe_data = temp
+
+        if res_class_dset is None or res_class_bins is None:
             res_class_bins = [None]
 
-        if cf_dset in gen.datasets:
-            cf_data = gen[cf_dset]
-        else:
-            cf_data = None
-            w = ('Could not find cf dataset "{}" in '
-                 'generation file: {}. Available datasets: {}'
-                 .format(cf_dset, gen_fpath, gen.datasets))
-            logger.warning(w)
-            warn(w, OutputWarning)
-
-        if lcoe_dset in gen.datasets:
-            lcoe_data = gen[lcoe_dset]
-        else:
-            lcoe_data = None
-            w = ('Could not find lcoe dataset "{}" in generation file: {} or '
-                 'econ file: {}. Available datasets: {}'
-                 .format(lcoe_dset, gen_fpath, econ_fpath, gen.datasets))
-            logger.warning(w)
-            warn(w, OutputWarning)
+        lcoe_recalc_req = ('fixed_charge_rate', 'capital_cost',
```

Comment:
I'd think it would help to move all of this logic into its own method. Its a lot and a dedicated docstring explaining the difference between missing, missing1, and missing2 would go a long way.

Reply:
I disagree. This is a single purpose method that will only be called once. Everything in the method is in the spirit of a single purpose - to check and retrieve input data. Besides, a method docstring won't describe the internal "missing" variables. I think the real solution here is just to better name the "missing" variables?

Reply:
That works too. I still think you could have a checking method inside the retrieval method, BUT as you said the important part is making the logic easier to follow. Renaming the internal missing variables should accomplish that.

Reply:
As our code base continues to grow, i think we really need to think harder about the method chaining. This is a great example. It almost hits the complexity limit, but as far as code design goes, it is a totally appropriate single-purpose method. There are some huge benefits to smaller methods with good doc strings, but there's also a huge challenge in chasing an object as it gets passed down through a billion levels of nested methods. As the zen of python says: "Flat is better than nested"

Reply:
I'm super torn on this one. I definitely agree with avoiding unnecessary method chaining, BUT I also get the need to limit complexity. I stumbled upon this article this weekend about [refactoring legacy code](https://opensource.com/article/21/7/legacy-code ) and his complex if statement example is what made me suggest a sub-method here.

Reply:
Yes I think neither extreme is appropriate. But here I think we have a good example of narrow purpose method that needs to have a somewhat higher complexity. We could break it up into a bunch of different methods but that would only add depth and i dont think it would add convenience. From that article, we are VERY far from the statement "If you cram all capabilities an object can offer into a single, giant function"